spec/models/course_spec.rb

```diff
@@ -142,9 +142,457 @@
     end
   end
 
-  describe '#upload_assignment_list' # TODO
-  describe '#get_required_files' # TODO
+  describe '#upload_assignment_list' do
+    let!(:test_course) do
+      Course.where(name: 'TestCourse-2022-Fall', display_name: 'Test Course')
+            .first_or_create(name: 'TestCourse-2022-Fall', is_hidden: false,
+                             display_name: 'Test Course')
+    end
+    before(:each) do
```

Comment:
This block shouldn't be necessary, as database objects should be cleared in between each test. (If there were other errors in your test cases, this isn't the way to fix thehm.)